test/test.js

```diff
@@ -1319,6 +1319,20 @@ describe('node-fetch', () => {
 		});
 	});
 
+	it('should create http.Agent class if options.agent is provided as an object', function() {
```

Comment:
yes, we can add an https specific test. 

I didn't think it was necessary since `https.Agent` inherits `http.Agent` and the logic is mainly concerned with initializing the object internally.

can add a test later today.